carbon-design-system/carbon
# test(project): add e2e test support with playwright

This PR adds an e2e testing harness for Visual Regression Testing with Percy & Playwright. It also removes support for Cypress and updates the VRT job in our CI workflow to use Playwright instead.

At a high-level, this change will run tests in CI that will report components in different themes to Percy which will then handle the image diff'ing. These tests are sharded in order to run in parallel and will correctly correspond to the same build for a PR. 

#### Changelog

**New**

- Add support for Playwright testing in the project
- Add testing guide in `testing.md`

**Changed**

- Update our VRT workflow to no longer call Cypress and instead call Playwright

**Removed**

- Remove support for Cypress in `packages/react`

#### Testing / Reviewing

- Verify that the two runs for the VRT job are reporting to the same percy build as reported in status checks
- Verify that screenshots are correctly generated
- Read through the testing guide, try out commands, see if it makes sense 👀
